src/riscv.ts

```diff
@@ -0,0 +1,584 @@
+    for (const regName of this.regList) {
+      const regVal = this.panicInfo.regs[regName] || 0;
+      const regBytes = Buffer.alloc(4);
+      regBytes.writeUInt32LE(regVal);
+      const regValHex = regBytes.toString('hex');
+      response += regValHex;
+    }
+    this.debug(`Register response: ${response}`);
+    this._respond(response, socket);
+  }
+
+  private _respondMem(startAddr: number, size: number, socket: net.Socket) {
+    const stackAddrMin = this.panicInfo.stackBaseAddr;
+    const stackData = this.panicInfo.stackData;
+    const stackLen = stackData.length;
+    const stackAddrMax = stackAddrMin + stackLen;
+
+    const inStack = (addr: number) =>
+      stackAddrMin <= addr && addr < stackAddrMax;
+
+    let result = '';
+    for (let addr = startAddr; addr < startAddr + size; addr++) {
+      if (!inStack(addr)) {
+        result += '00';
+      } else {
+        result += stackData[addr - stackAddrMin].toString(16).padStart(2, '0');
+      }
+    }
+
+    this._respond(result, socket);
+  }
+}
+
+const exceptions = [
+  { code: 0x0, description: 'Instruction address misaligned' },
+  { code: 0x1, description: 'Instruction access fault' },
+  { code: 0x2, description: 'Illegal instruction' },
+  { code: 0x3, description: 'Breakpoint' },
+  { code: 0x4, description: 'Load address misaligned' },
+  { code: 0x5, description: 'Load access fault' },
+  { code: 0x6, description: 'Store/AMO address misaligned' },
+  { code: 0x7, description: 'Store/AMO access fault' },
+  { code: 0x8, description: 'Environment call from U-mode' },
+  { code: 0x9, description: 'Environment call from S-mode' },
+  { code: 0xb, description: 'Environment call from M-mode' },
+  { code: 0xc, description: 'Instruction page fault' },
+  { code: 0xd, description: 'Load page fault' },
+  { code: 0xf, description: 'Store/AMO page fault' },
+];
+
+type RiscvFQBN = FQBN & { boardId: Target };
+
+function isRiscvFQBN(fqbn: FQBN): fqbn is RiscvFQBN {
+  return isTarget(fqbn.boardId);
+}
+
+function buildPanicServerArgs(elfPath: string, port: number): string[] {
+  return [
+    '--batch',
+    '-n',
+    elfPath,
+    // '-ex', // executes a command
+    // `set remotetimeout ${debug ? 300 : 2}`, // Set the timeout limit to wait for the remote target to respond to num seconds. The default is 2 seconds. (https://sourceware.org/gdb/current/onlinedocs/gdb.html/Remote-Configuration.html)
+    '-ex',
+    `target remote :${port}`, // https://sourceware.org/gdb/current/onlinedocs/gdb.html/Server.html#Server
+    '-ex',
+    'bt',
+  ];
+}
+
+async function processPanicOutput(
+  params: DecodeParams,
+  panicInfo: PanicInfo,
+  options: DecodeOptions
+): Promise<string> {
+  const { elfPath, toolPath } = params;
+  let server: { close: () => void } | undefined;
+  try {
+    const gdbServer = new GdbServer({
+      panicInfo,
+      debug: options.debug,
+    });
+    const { port } = await gdbServer.start();
```

Comment:
`start` could accept a `signal` and not wait for the `connect` event. It would be more responsive